runtime/sema/check_for.go

```diff
@@ -123,3 +99,68 @@ func (checker *Checker) VisitForStatement(statement *ast.ForStatement) (_ struct
 
 	return
 }
+
+func (checker *Checker) loopVariableType(valueType Type, hasPosition ast.HasPosition) Type {
```

Comment:
How does this differ from the type checking result of indexing? Maybe the existing indexing checking can be re-used here for the loop variable checking?